refactor: reimplement csv option for domains

erika-wallace opened the pull request:
## Description
[W-19830244](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00002MzpbAYAR/view)

This PR reimplements `--csv` for `heroku domains`, since it was removed from oclif during the migration to v4. It allows users to output domain data in CSV format and export to a `.csv` file.

### Type of change:
Refactor

## Testing
**NOTE:** You will need an app with multiple domains (at least 3-4 to test sorting).

**Test CSV output:**
1. checkout this branch and run `npm install && npm run build`
2. Run `./bin/run domains -a <your-app> --csv`, confirm CSV output with headers
3. Run `./bin/run domains -a <your-app> --csv > test-output.csv`, confirm file was created successfully and opens with proper CSV format

**Test CSV sorting:**

4. Run `./bin/run domains -a <your-app> --csv --sort "Domain Name"`, confirm CSV output was sorted by the Domain Name column (direct property)
5. Run `./bin/run domains -a <your-app> --csv --sort "DNS Record Type"`, confirm CSV output was sorted by the DNS Record Type column (not a direct property)

**Test CSV with other flags:**

6. Run `./bin/run domains -a <your-app> --csv --extended`, confirm output includes additional columns (ACM Status, ACM Status Reason)
7. Run `./bin/run domains -a <your-app> --csv --columns "Domain Name,DNS Target"`, confirm output only shows specified columns
8. Run `./bin/run domains -a <your-app> --csv --filter "Domain Name=example"`, confirm output only shows domains matching the filter criteria

**Verify no regressions:**

9. Run `./bin/run domains -a <your-app>`, confirm table output still works
10. Run `./bin/run domains -a <your-app> --sort "Domain Name"`, confirm table sorting still works

erika-wallace commented:
**Note:** The sorting for CSV is different from the table sorting, so the outputs don't match exactly. The table is sorted by oclif/table, which uses the `natural-orderby` package. Let me know if we should align them. I can import `natural-orderby` for CSV too.

zwhitfield3 commented on packages/cli/src/commands/domains/index.ts:
```diff
@@ -173,8 +200,8 @@ www.example.com  CNAME            www.example.herokudns.com
-        if (customDomains.length > 100 && !flags.json) {
-          ux.warn(`This app has over 100 domains. Your terminal may not be configured to display the total amount of domains. You can output domains in JSON format with: ${color.cmd('heroku domains -a example-app --json')}`)
+        if (customDomains.length > 100 && !flags.json && !flags.csv) {
+          ux.warn(`This app has over 100 domains. Your terminal may not be configured to display the total amount of domains. You can export all domains into a CSV file with: ${color.cmd('heroku domains -a example-app --csv > example-file.csv')}`)
```
We should leave the JSON copy as is because that was the initial flow for users with over 100 domains. The previous functionality still had CSV formatting available, but wasn't highlighted in the warning. They also can view that format option in the `--help` CSV flag description.

zwhitfield3 commented on packages/cli/src/commands/domains/index.ts:
```diff
@@ -150,6 +151,32 @@ www.example.com  CNAME            www.example.herokudns.com
 
+  outputCSV = (customDomains: Heroku.Domain[], tableConfig: Record<string, any>, sortProperty?: string) => {
+    const getValue = (domain: Heroku.Domain, key: string, config?: Record<string, any>) => {
+      const cfg = config ?? tableConfig[key]
```
Can we make `cfg` a little more descriptive? Like `tableConfigValue` or `columnValue`? This will help us maintain readability.

erika-wallace commented:
@zwhitfield3 I updated the code with your suggestions and added `natural-orderby` as a dependency to maintain consistency of sorting between the table and CSV output
